app/uk/gov/hmrc/helptosavefrontend/controllers/BankAccountController.scala

```diff
@@ -77,31 +76,30 @@ class BankAccountController @Inject() (val helpToSaveService: HelpToSaveService,
       session ⇒
         BankDetails.giveBankDetailsForm().bindFromRequest().fold(
           withErrors ⇒
-            Ok(views.html.register.bank_account_details(withErrors, backLinkFromSession(session))),
-          { bankDetails ⇒
-            barsService.validate(htsContext.nino, bankDetails, routes.BankAccountController.submitBankDetails().url).map[Future[Result]] {
-              case Right(true) ⇒
-                sessionStore.store(session.copy(bankDetails = Some(bankDetails)))
-                  .fold(
-                    error ⇒ {
-                      logger.warn(s"Could not update session with bank details: $error")
-                      internalServerError()
-                    },
-                    _ ⇒ SeeOther(routes.RegisterController.getCreateAccountPage().url)
-                  )
+            Ok(views.html.register.bank_account_details(withErrors, backLinkFromSession(session))), { bankDetails ⇒
+            helpToSaveService.validateBankDetails(ValidateBankDetailsRequest(htsContext.nino, bankDetails.sortCode.toString, bankDetails.accountNumber)).fold[Future[Result]](
+              error ⇒ {
+                logger.warn(s"Could not validate bank details due to : $error")
+                internalServerError()
+              }, { isValid ⇒
+                if (isValid) {
+                  sessionStore.store(session.copy(bankDetails = Some(bankDetails)))
+                    .fold(
+                      error ⇒ {
+                        logger.warn(s"Could not update session with bank details: $error")
+                        internalServerError()
+                      },
+                      _ ⇒ SeeOther(routes.RegisterController.getCreateAccountPage().url)
+                    )
+                } else {
+                  val formWithErrors = BankDetails.giveBankDetailsForm().fill(bankDetails)
+                    .withError("sortCode", BankDetailsValidation.ErrorMessages.sortCodeBarsInvalid)
```

Comment:
we've still got a reference to BARS here and the line below

Reply:
i thought these are specific to bars service -  let me take a closer look

Reply:
fixed